torch/csrc/jit/passes/onnx/pattern_conversion/autograd_function_process.cpp

```diff
@@ -0,0 +1,62 @@
+#include <torch/csrc/jit/passes/onnx/pattern_conversion/autograd_function_process.h>
+
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/onnx/helper.h>
+#include <torch/csrc/jit/passes/utils/subgraph_utils.h>
+
+namespace torch {
+namespace jit {
+
+void convertSubgraphToSubBlock(Block* block) {
+  for (auto it = block->nodes().begin(), end = block->nodes().end();
+       it != end;) {
+    Node* node = *it++;
+    for (auto block : node->blocks()) {
+      convertSubgraphToSubBlock(block);
+    }
+
+    if (node->kind() == prim::PythonOp) {
+      // Construct subblock
+      auto subblock = node->addBlock();
+      auto graph = subblock->owningGraph();
+
+      std::unordered_map<Value*, Value*> env;
+      // Populate subblock with subgraph nodes
+      auto subgraph = node->g(attr::Subgraph);
+      for (const auto i : c10::irange(subgraph->inputs().size())) {
+        subblock->addInput()->copyMetadata(subgraph->inputs()[i]);
+        env[subgraph->inputs()[i]] = subblock->inputs()[i];
+      }
+      for (auto* n : subgraph->nodes()) {
+        auto cloned_n =
+            subblock->appendNode(graph->createClone(n, [&](Value* v) {
+              return env.find(v) != env.end() ? env[v] : v;
+            }));
+        for (size_t i = 0; i < n->outputs().size(); ++i) {
+          env[n->outputs().at(i)] = cloned_n->outputs().at(i);
+          auto it = std::find(
```

Comment:
move this out of `for (auto* n : subgraph->nodes())`, and iterate through `subgraph->outputs()` to register new outputs for `subblock`.